### ISSUE TITLE
Compose: Hide MixMaster chain

### ISSUE THREAD

**Issue by flatcap** [neomutt/neomutt#4216]
## Overview

Mixmaster is an anonymising proxy for email.
It's an old technology, but it still works.

It doesn't add any build dependencies to NeoMutt, so some packagers choose to enable it.

Typical Compose Dialog:
```
    From:
      To:
      Cc:
     Bcc:
 Subject:
Reply-To:
     Fcc:
Security: None
     Mix: <no chain defined>
```

For most users, the 'Mix' line is just clutter.

## Task: Display the 'Mix' line on demand

The Compose Dialog is composed of an Envelope and an Attachment List.

The Envelope is a self-contained Window. (`envelope/window.c`)
`MuttWindow`s in NeoMutt implement two API functions:
- `recalc()`
  Perform any (slow) work needed to generate the display data
- `repaint()`
  Display the data (quickly)

Mixmaster stores the chain in `Email.chain`

**Tasks**:
- Alter `env_recalc()` to depend on `Email.chain`
- Alter `env_repaint()` to depend on `Email.chain`

## Testing

To test this work, it's not necessary to use MixMaster itself.

Put this script on your path and make it executable.
It will generate some realistic data for the MixMaster Dialog.

- https://raw.githubusercontent.com/neomutt/test-rigs/main/mixmaster

To create a MixMaster Chain, starting in the Compose Dialog:

- Hit <kbd>M</kbd> (`<mix>`)
- Select a few hosts using <kbd>\<space\></kbd>.
  They'll show up in the Chain window below.
- Hit <kbd>q</kbd> (`<exit>`) to leave

The Envelope will now display something like:
```
Mix: Cherry, Damson, Elderberry, Fig
```

---

Any questions, just ask.
Thanks!

**Comment by Pierre-Colin** [comment]
I reckon that by `Email.chain` you mean `wdata->email->chain` in both functions.

By displaying the Mix line “on demand,” do you mean simply not printing it when it’s empty instead of `<no chain defined>`? If so, I know approximately what to do.

**Comment by flatcap** [comment]
> I reckon that by `Email.chain` you mean `wdata->email->chain` in both functions.

Yes.  I referred to the source of the data: `struct Email` member `chain`.

In the code, the Envelope attaches some private data (`EnvelopeWindowData`) to `struct MuttWindow` (that's the `wdata`).
This holds the state of the Window and includes the `struct Email` that we're composing.

> By displaying the Mix line “on demand,” do you mean simply not printing it when it’s empty instead of `<no chain defined>`?

Exactly.

> I know approximately what to do

Great :-)
Thanks!

**Comment by flatcap** [comment]
Fixed in #4220
